### DISCUSSION TITLE
How to enable and use  mirrord in  Intellije availe remotely on WSL2

### DISCUSSION THREAD

**Discussion by SanadhyaLakshya** [metalbear-co/mirrord#2375] [Q&A]
I have a service and several pods deployed via a Helm chart in a WSL2 environment. I need to redirect all incoming traffic from one specific pod to a codebase running in IntelliJ IDEA, where I have installed the Mirrord plugin. However, I am encountering an issue when i enable the mirrord plugin When attempting to start the server, I receive the following error message: "Application Server was not connected before run configuration stop, reason: Unable to ping server at localhost:1099.".. when i disable the mirrod plugin the service is deployed properly on tomcat ..
`This is my mirrord config: 
{
    "kubeconfig": "/home/ls/.kube/config",
    "feature": {
        "network": {
            "incoming": "steal",
            "outgoing": true
        },
        "fs": "read",
        "env": true
    },
    "target": {
        "path": "pod/portal-service-0",
        "namespace": "iem"
    },
    "agent": {
        "namespace": "iem"
    }
}`

and my tomcat configuration is attached in image 
![2024-04-16 14_45_05-com mwp p – mirrord json](https://github.com/metalbear-co/mirrord/assets/25980789/a7689d17-d95d-435b-901d-abe76312b2bd)

How can steal all the incoming and outcoming traffic from mirrord.. while debuug the java app locally in tomcat? Could you please Suggest?

**Comment by aviramha** [comment]
Hi! Sorry about the troubles. Can you try the following mirrord config file?
```json
{
  "kubeconfig": "/home/ls/.kube/config",
  "feature": {
    "network": {
      "incoming": {
        "mode": "steal",
        "ignore_ports": [1099]
      },
      "outgoing": {
        "filter": {
          "local": [":1099"]
        }
      }
    },
    "fs": "read",
    "env": true
  },
  "target": {
    "path": "pod/portal-service-0",
    "namespace": "iem"
  },
  "agent": {
    "namespace": "iem"
  }
}
```

**Reply by SanadhyaLakshya** [reply]
I successfully deployed the Java code using the specified configuration in IntelliJ IDEA installed in WSL environment, and it is currently operational at http://localhost:8080/p.service/. However, the interception/steal of the target pod is not taking place as expected. any other setting or configuration required..

**Reply by aviramha** [reply]
is the remote service also listening on port 8080? if not, you'd either need to listen on same port or use `port_mapping` configuration under `incoming` to map between local and remote ports.

**Comment by SanadhyaLakshya** [comment]
Yes the remote service is listening on the same port .. ie the tomcat is running on the same port in remote pod
![2024-04-18 18_09_14-Select ls@DI2INUDR0076WNB_ ~](https://github.com/metalbear-co/mirrord/assets/25980789/3d4bea1d-8476-4de7-90af-15c9ff602b15)

Mirror d agent is also running 
![2024-04-18 18_12_30-Select ls@DI2INUDR0076WNB_ ~](https://github.com/metalbear-co/mirrord/assets/25980789/0dea6d24-602d-4476-bf74-955892cf2c25)

**Reply by aviramha** [reply]
Sorry, the important thing is the pod port, can you check that ?

**Reply by SanadhyaLakshya** [reply]
I think its the pod port only..  we have passed it while deployment
```json
apiVersion: v1
kind: Service
metadata:
  name: portal-service
  namespace: iem
spec:
  ports:
    - name: http-pservice
      appProtocol: http
      protocol: TCP
      port: 8080
      targetPort: 8080
  selector:
    run: portal-service
```

**Reply by aviramha** [reply]
That looks like it should work. Would you like to do a screen share and debug it?
